libs/designsystem/src/lib/components/slides/slides.component.ts

```diff
@@ -0,0 +1,76 @@
+import { EventEmitter } from '@angular/core';
+import {
+  Component,
+  ChangeDetectionStrategy,
+  Input,
+  ContentChild,
+  TemplateRef,
+  Directive,
+  AfterViewInit,
+  ViewChild,
+  ChangeDetectorRef,
+  Output,
+} from '@angular/core';
+import { IonSlides } from '@ionic/angular';
+
+@Directive({
+  selector: '[kirbySlide]',
+})
+export class SlideDirective {}
+
+@Component({
+  selector: 'kirby-slides',
+  template: `
+    <ion-slides [options]="slidesOptions" #ionslides (ionSlideDidChange)="onSlideChanged($event)">
+      <ion-slide *ngFor="let slide of slides" [ngClass]="{ 'highlight-active': onlyOneSlideShown }">
+        <ng-container
+          *ngTemplateOutlet="slideTemplate; context: { $implicit: slide }"
+        ></ng-container>
+      </ion-slide>
+    </ion-slides>
+  `,
+  styleUrls: ['./slides.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SlidesComponent implements AfterViewInit {
+  @ViewChild('ionslides', { static: false }) ionSlides: IonSlides;
+  @Input() slidesOptions: any;
+  @Input() slides: any[];
+
+  @Output() selectedRecord = new EventEmitter<any>();
```

Comment:
consider renaming to something like: slideChange, selectedRecord is not really "event like"

Reply:
already done...